glowfic-dl.py

```diff
@@ -207,13 +308,30 @@ async def download_chapter(
     return (title, list(render_posts(posts, image_map, authors)))
 
 
-def compile_chapters(chapters: list[tuple[str, list[Section]]]) -> list[epub.EpubHtml]:
+def compile_chapters(
+    chapters: list[tuple[str, list[Section]]]
+) -> Iterable[list[epub.EpubHtml]]:
+    chapter_digits = len(str(len(chapters)))
     anchor_sections = {}
+
+    # Map permalinks to file names
     for (i, (title, sections)) in enumerate(chapters):
+        section_digits = len(str(len(sections)))
         for (j, section) in enumerate(sections):
-            file_name = "chapter%i_%i.html" % (i, j)
+            file_name = "Text/" + make_filename_valid_for_epub3(
```

Comment:
Doesn't "Text/" cut into the 255 character limit?

Reply:
It doesn't. The 255-byte limit is on each path component, not on the full path. (There's a 65535-byte limit on total path length which "Text/" _does_ cut into, but there's currently no risk of colliding with that one, since everything but the final filename is hardcoded.)